lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/util/TestingHttpServer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.sun.net.httpserver.HttpServer;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+
+public class TestingHttpServer
```

Comment:
Can we copy the implementation from `ExampleHttpServer` - so that we don't have to add a response. We could have a servlet which use the file from resources.

Reply:
I added new TestHttpServer which is based on ExampleHttpServer. It is thread-safe now and can be used by concurrent test methods